cmd/metal-api/internal/service/partition-service.go

```diff
@@ -206,13 +206,23 @@ func (r *partitionResource) createPartition(request *restful.Request, response *
 	}
 
 	var dnsServers metal.DNSServers
-	if len(requestPayload.DNSServers) > 0 {
-		dnsServers = requestPayload.DNSServers
+	reqDNSServers := requestPayload.DNSServers
+	if len(reqDNSServers) > 0 {
```

Comment:
i think this check can be omited, because the validate func iterates over the dnsservers otherwise returns nil